presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/DesugarTryExpression.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.planner.iterative.rule;
+
+import com.facebook.presto.sql.planner.DesugarTryExpressionRewriter;
+import com.facebook.presto.sql.planner.iterative.Rule;
+import com.facebook.presto.sql.planner.iterative.RuleSet;
+import com.facebook.presto.sql.tree.Expression;
+
+import java.util.Set;
+
+public class DesugarTryExpression
+        implements RuleSet
+{
+    @Override
+    public Set<Rule<?>> rules()
+    {
+        ExpressionRewriteRuleSet ruleSet = new ExpressionRewriteRuleSet(DesugarTryExpression::rewrite);
```

Comment:
I am not familiar with this part, but I see `CanonicalizeExpressions` directly extends `ExpressionRewriteRuleSet`.

Can we also do this for `DesugarTryExpression` as well? @kokosing